services/blockchain-indexer/shared/dataService/business/generators.js

```diff
@@ -13,46 +13,23 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
-// const BluebirdPromise = require('bluebird');
-const { Utils } = require('lisk-service-framework');
-const {
-	// getIndexedAccountInfo,
-	getBase32AddressFromHex,
-} = require('../../utils/accountUtils');
+const BluebirdPromise = require('bluebird');
 
+const { getBase32AddressFromHex } = require('../../utils/accountUtils');
 const { getGenesisConfig } = require('../../constants');
-
 const { requestConnector } = require('../../utils/request');
-
-const ObjectUtilService = Utils.Data;
-const { isProperObject } = ObjectUtilService;
+const { getNameByAddress } = require('../../utils/delegateUtils');
 
 const getGenerators = async () => {
-	const generators = await requestConnector('getGenerators');
-	generators.data = generators
-		.map(generator => ({
-			...generator,
-			address: getBase32AddressFromHex(generator.address),
+	const { list: generatorsAddresses } = await requestConnector('getGenerators');
+	const generators = await BluebirdPromise.map(
+		generatorsAddresses,
+		async address => ({
+			address: getBase32AddressFromHex(address),
+			name: await getNameByAddress(address),
```

Comment:
Please also include `nextForgingTime` with a `TODO` comment, if the data is not currently available. Please check with the SDK team for the same.